Increase ping timeout for registries

dmcgowan opened the pull request:
Ensure v2 registries are given more than 5 seconds to return a ping and avoid an unnecessary fallback to v1.

Some v1 only registries may experience the additional 10 second delay if the `/v2` endpoint is not returning a result.

icecrime commented:
LGTM with a small thing: can we please make sure there's a log on the daemon-side when we hit a timeout while trying to ping? Thanks!

dmcgowan commented:
This would already get logged
```
DEBU[0017] Error getting v2 registry: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection
```

calavera commented:
@dmcgowan it looks like that's debug level, isn't it? I think what @icecrime is asking is for an error at the error level that say something like `request timed out...`

dmcgowan committed:
Increase ping timeout

Ensure v2 registries are given more than 5 seconds to return a ping and avoid an unnecessary fallback to v1.
Elevates log level about failed v2 ping to a warning to match the warning related to using v1 registries.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

dmcgowan commented:
Since we are now warning when using a v1 registry it would make sense to elevate this message from Debug to Warn. Checking whether the request failed because it was a timeout would require a larger code change since the error type does not indicate whether a timeout occurred. The hidden `http.httpError` type does have a `Timeout() bool` function which can be tested for, however in both the cases I mentioned above the error is actually a `url.Error` with no way of extracting whether it occurred as a result of a timeout. The correct way to do this is through supporting the `CancelRequest` method as suggested by `http.Client`.

calavera commented:
I'm okay with this warning.

LGTM
